python_modules/dagster/dagster/_core/definitions/asset_out.py

```diff
@@ -113,44 +68,135 @@ def __new__(
         tags: Optional[Mapping[str, str]] = None,
         **kwargs,
     ):
+        """Defines an asset produced by a :py:func:`@multi_asset <multi_asset>`.
+
+        Args:
+            key_prefix (Optional[Union[str, Sequence[str]]]): If provided, the asset's key is the
+                concatenation of the key_prefix and the asset's name. When using ``@multi_asset``, the
+                asset name defaults to the key of the "outs" dictionary Only one of the "key_prefix" and
+                "key" arguments should be provided.
+            key (Optional[Union[str, Sequence[str], AssetKey]]): The asset's key. Only one of the
+                "key_prefix" and "key" arguments should be provided.
+            dagster_type (Optional[Union[Type, DagsterType]]]):
+                The type of this output. Should only be set if the correct type can not
+                be inferred directly from the type signature of the decorated function.
+            description (Optional[str]): Human-readable description of the output.
+            is_required (bool): Whether the presence of this field is required. (default: True)
+            io_manager_key (Optional[str]): The resource key of the IO manager used for this output.
+                (default: "io_manager").
+            metadata (Optional[Dict[str, Any]]): A dict of the metadata for the output.
+                For example, users can provide a file path if the data object will be stored in a
+                filesystem, or provide information of a database table when it is going to load the data
+                into the table.
+            group_name (Optional[str]): A string name used to organize multiple assets into groups. If
+                not provided, the name "default" is used.
+            code_version (Optional[str]): The version of the code that generates this asset.
+            freshness_policy (Optional[FreshnessPolicy]): (Deprecated) A policy which indicates how up
+                to date this asset is intended to be.
+            automation_condition (Optional[AutomationCondition]): AutomationCondition to apply to the
+                specified asset.
+            backfill_policy (Optional[BackfillPolicy]): BackfillPolicy to apply to the specified asset.
+            owners (Optional[Sequence[str]]): A list of strings representing owners of the asset. Each
+                string can be a user's email address, or a team name prefixed with `team:`,
+                e.g. `team:finops`.
+            tags (Optional[Mapping[str, str]]): Tags for filtering and organizing. These tags are not
+                attached to runs of the asset.
+        """
+        # Accept a hidden "spec" argument to allow for the AssetOut to be constructed from an AssetSpec
+        # directly. This is used in the AssetOut.from_spec method.
+        spec = kwargs.get("spec")
+        if spec:
+            del kwargs["spec"]
```

Comment:
why del?

Reply:
(alternative is to make it a real hidden param and set emit_runtime_warning to false

Reply:
just to bypass `only_allow_hidden_params_in_kwargs(AssetOut, kwargs)` from yelling at us, since `spec` is not a `hidden_param` (which we have only set up for deprecated hidden params rather than hidden "system"/framework params)